src/client/auth/mod.rs

```diff
@@ -122,6 +122,18 @@ impl AuthMechanism {
                 };
                 Ok(())
             }
+            #[cfg(feature = "tokio-runtime")]
+            AuthMechanism::MongoDbAws => {
+                if credential.username.is_some() && credential.password.is_none() {
+                    return Err(ErrorKind::ArgumentError {
+                        message: "Username cannot be provided with password for MONGODB-AWS \
+                                  authentication"
+                            .to_string(),
+                    }
+                    .into());
+                }
+                Ok(())
+            }
```

Comment:
Is an AWS credential without a username valid?

Reply:
Yes, as long as there's also no password. In addition to the user specifying them, the driver also can obtain them through [environmental variables or ECS/EC2 endpoints](https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#obtaining-credentials).